lib/hllcounter.cc

```diff
@@ -307,68 +222,69 @@ void HLLCounter::set_erate(double error_rate)
         throw InvalidValue("Please set error rate to a value "
                            "greater than zero");
     }
-    int p = ceil(log2(pow(1.04 / error_rate, 2)));
-    this->init(p, this->_ksize);
+    int new_p = ceil(log2(pow(1.04 / error_rate, 2)));
+    init(1 << new_p, _ksize);
 }
 
 void HLLCounter::set_ksize(WordLength new_ksize)
 {
-    if (count(this->M.begin(), this->M.end(), 0) != this->m) {
+    if (count(begin(counters), end(counters), 0) != ncounters) {
         throw ReadOnlyAttribute("You can only change k-mer size prior to "
                                 "first counting");
     }
 
-    this->init(this->p, new_ksize);
+    _ksize = new_ksize;
 }
 
 double HLLCounter::_Ep()
 {
-    double sum = accumulate(this->M.begin(), this->M.end(), 0.0, ep_sum);
-    double E = this->alpha * pow(this->m, 2.0) / sum;
+    double sum = 0.0;
+    for (auto v: counters) {
+        sum += pow(2.0, float(-v));
+    }
 
-    if (E <= (5 * (double)this->m)) {
-        return E - estimate_bias(E, this->p);
+    double E = alpha * pow(ncounters, 2.0) / sum;
+
+    if (E <= (5 * (double)ncounters)) {
+        return E - estimate_bias(E, ncounters_log2);
     }
 
     return E;
 }
 
 uint64_t HLLCounter::estimate_cardinality()
 {
-    long V = count(this->M.begin(), this->M.end(), 0);
+    long V = count(begin(counters), end(counters), 0);
 
     if (V > 0) {
-        double H = this->m * log((double)this->m / V);
-        if (H <= get_threshold(this->p)) {
+        double H = ncounters * log((double)ncounters / V);
+        if (H <= get_threshold(ncounters_log2)) {
             return H;
         }
     }
-    return this->_Ep();
+    return _Ep();
 }
 
 void HLLCounter::add(const std::string &value)
 {
     HashIntoType x = khmer::_hash_murmur(value);
-    uint64_t j = x & (this->m - 1);
-    this->M[j] = std::max(this->M[j], get_rho(x >> this->p, 64 - this->p));
+    uint64_t j = x & (ncounters - 1);
+    counters[j] = std::max(counters[j],
+                           get_rho(x >> ncounters_log2, 64 - ncounters_log2));
 }
 
-unsigned int HLLCounter::consume_string(const std::string &inp)
+unsigned int HLLCounter::consume_string(const std::string &s)
```

Comment:
Also matches the test here https://github.com/dib-lab/khmer/pull/1511/files/32fcf25137937de2dc52d214724f87312434c51a#diff-8e86ee1950f0fb1c11b9da2d3a59a088R193 :)